### ISSUE TITLE
Is it possible to get the remote IP in crow::request?

### ISSUE THREAD

**Issue by sgiurgiu** [ipkn/crow#146]
As far as I could tell the information is available in the Adaptors (SocketAdaptor and SSLAdaptor) but it gets lost. Is it possible to somehow make it available?

Thank you.

**Comment by chengjianxi** [comment]
the same question

**Comment by mrozigor** [comment]
Maybe try to add std::string field (e.g. std::string ipAddress) and then set it in _http_connection.h:259_ with something like:

```c++
req.ipAddress = adaptor_.socket_.remote_endpoint().address().to_string();
```

Then you could use it via crow::request. Didn't test it, just looked into code.

**Comment by cadubentzen** [comment]
Hi @mrozigor 

> req.ipAddress = adaptor_.socket_.remote_endpoint().address().to_string();

I tested it and it works except for a typo (I think). This line worked for me (only remove `socket_`).

```cpp
req.ipAddress = adaptor_.remote_endpoint().address().to_string();
```

Do you have any interest to open a pull request for this?

**Comment by mrozigor** [comment]
Yeah, I want to make some improvements when I'll have some time ;) Try to make PR by the end of this week.

**Comment by mrozigor** [comment]
Added in #322. Changes are on my fork, on master branch.
